### PR TITLE
Non-trivial merge to main

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

**Review by rossabaker — COMMENTED** [review]
The hard merges were in things I don't understand. Help.

File: ember-server/shared/src/main/scala/org/http4s/ember/server/EmberServerBuilder.scala [hunk 1]
```diff
@@ -133,19 +139,53 @@ final class EmberServerBuilder[F[_]: Async] private (
...
+      deleteOnClose: Boolean = true) =
+    copy(unixSocketConfig = Some((unixSockets, unixSocketAddress, deleteIfExists, deleteOnClose)))
+  def withoutUnixSocketConfig =
+    copy(unixSocketConfig = None)
+
   def build: Resource[F, Server] =
     for {
       sg <- sgOpt.getOrElse(Network[F]).pure[Resource[F, *]]
-      ready <- Resource.eval(Deferred[F, Either[Throwable, SocketAddress[IpAddress]]])
+      ready <- Resource.eval(Deferred[F, Either[Throwable, SocketAddress[Host]]])
```
[the remark this answers is not in the document]
Comment (armanbilge) [reply ember-server/shared/src/main/scala/org/http4s/ember/server/EmberServerBuilder.scala:155]: This is my screw-up I think. When I changed `address` in server to use ip4s it should have been `SocketAddress[IpAddress]` not `SocketAddress[Host]`. Unless there's a reason to keep `Host` we should fix this there.

File: client/shared/src/main/scala/org/http4s/client/middleware/UnixSocket.scala [hunk 2]
```diff
@@ -14,6 +14,15 @@
  * limitations under the License.
  */
 
-package org.http4s.laws.discipline
+package org.http4s.client.middleware
```
[the remark this answers is not in the document]
Comment (armanbilge) [reply client/shared/src/main/scala/org/http4s/client/middleware/UnixSocket.scala:17]: It's these headers that dwarf the actual source ...

[the remark this answers is not in the document]
Comment (rossabaker) [reply client/shared/src/main/scala/org/http4s/client/middleware/UnixSocket.scala:17]: We used to use SPDY headers, but I wanted to stop being weird.

File: ember-server/shared/src/main/scala/org/http4s/ember/server/EmberServerBuilder.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (rossabaker) [reply ember-server/shared/src/main/scala/org/http4s/ember/server/EmberServerBuilder.scala:155]: I didn't know how to turn the Unix socket path into an IP address.

[the remark this answers is not in the document]
Comment (armanbilge) [reply ember-server/shared/src/main/scala/org/http4s/ember/server/EmberServerBuilder.scala:155]: Well, it doesn't ... 🤔

**Comment by armanbilge** [comment]
I'll fix up this merge 👍 

But, do you think we can wait for #5285 to join? It's going to be an ugly, horrible merge but will significantly cleanup main and minimize the diff. These current conflicts here are just droplets in that storm.

**Comment by rossabaker** [comment]
I've been trying to get a release train going, but I'm fine with whatever on 1.0.  Especially if someone else is doing the nasty merge! :smile:

**Comment by armanbilge** [comment]
Ah, well #5285 is targeting 0.23, which is why I was asking. It probably won't make the next release train? In which case I shouldn't wait for it here.

**Comment by rossabaker** [comment]
I've seen it but haven't opened it.  I'll try to look this afternoon.

**Comment by armanbilge** [comment]
Thanks! Also, there's no rush for that to make the next release, it's important to do it right.

**Comment by rossabaker** [comment]
Let's see how controversial it is and then we can schedule it.

**Comment by armanbilge** [comment]
Superseded by #5303

**Pull request closed without merging**